templates/cpp-template-default/CMakeLists.txt

```diff
@@ -111,7 +111,9 @@ endif()
 include_directories(
   /usr/local/include/GLFW
   /usr/include/GLFW
+  ${COCOS2D_ROOT}
```

Comment:
Both are needed as far as I can tell unless we modify the headers themselves. Here are the full details:

First create a new project with the cocos version in `HEAD`.

````shell
cocos new -l cpp extensionstest

````

Add the following block somewhere in HelloWorldScene::init():

````cpp
auto minusSprite = Sprite::create("HelloWorld.png");
auto plusSprite = Sprite::create("HelloWorld.png");
auto stepper = extension::ControlStepper::create(minusSprite, plusSprite);
stepper->setScale(.5);
stepper->setStepValue(.05);
stepper->setMinimumValue(1);
stepper->setMaximumValue(10);
stepper->setValue(1);
stepper->setPosition(visibleSize * 9.f / 10.f);
this->addChild(stepper);
````

After trying to compile (i.e. `cocos run -p linux -j$(nproc)`), you'll get:

````cpp
/path/to/extensionstest/Classes/HelloWorldScene.cpp:69:20: error: ‘extension’ has not been declared
auto stepper = extension::ControlStepper::create(minusSprite, plusSprite);
````

Looking at the directory tree from the root of the project, we get:

```shell
$ cd ${PROJECT_ROOT} && tree -d -L 3
.
├── Classes
├── cocos2d
│   ...
│   ├── extensions
│   │   ├── assets-manager
│   │   ├── GUI
│   │   ├── Particle3D
│   │   └── physics-nodes
````

which means that including the following in `HelloWorldScene.cpp`, without modifying the `CMakeLists.txt`, should work:

```cpp
#include <../extensions/cocos-ext.h>
````

You'll get the following error:

````cpp
/path/to/extensionstest/cocos2d/cocos/../extensions/GUI/CCControlExtension/CCControlUtils.h:39:40: fatal error: extensions/ExtensionExport.h: No such file or directory
 #include "extensions/ExtensionExport.h"
````

And this is why we need to include `${COCOS2D_ROOT}` to the `include_directories` list in `CMakeLists.txt`.

Fixing the `#include` in `HelloWorldScene.cpp` to the following:

```cpp
#include <extensions/cocos-ext.h>
````

Adding `${COCOS2D_ROOT}` to `CMakeLists.txt`, and recompiling, you'll get the following error:

````cpp
/path/to/extensionstest/cocos2d/cocos/../extensions/assets-manager/Manifest.h:37:35: fatal error: json/document-wrapper.h: No such file or directory
 #include "json/document-wrapper.h"
````

Looking at the directory tree again:

````shell
$ cd ${PROJECT_ROOT} && tree -d -L 3
.
├── Classes
├── cocos2d
│   ...
│   ├── external
│   │   ├...
│   │   ├── json
````

It becomes clear that we also have to add `${COCOS2D_ROOT}/external` to the `include_directories` in `CMakeLists.txt` because, again, the headers under the `external` directory assume that `${COCOS2D_ROOT}/external` has been added to the `include_directories` in `CMakeLists.txt` but that's not the case. After including `${COCOS2D_ROOT}/external` in `CMakeLists.txt` as in this pull request and recompiling once again, the scene will run.